fili-core/src/main/java/com/yahoo/bard/webservice/data/config/metric/makers/AggregationAverageMaker.java

```diff
@@ -115,80 +120,69 @@ private TemplateDruidQuery buildOuterQuery(
      * @return A template query representing the inner aggregation
      */
     private TemplateDruidQuery buildInnerQuery(MetricField sourceMetric, TemplateDruidQuery innerDependentQuery) {
-
         Set<Aggregation> newInnerAggregations = convertSketchesToSketchMerges(innerDependentQuery.getAggregations());
-        Set<PostAggregation> newInnerPostAggregations = new LinkedHashSet<>();
 
-        // If the sourceMetric is a Post Aggregator, we need to preserve it in the inner query
-        if (sourceMetric instanceof PostAggregation) {
-            newInnerPostAggregations.add((PostAggregation) sourceMetric);
-        }
+        Set<PostAggregation> newInnerPostAggregations = !(sourceMetric instanceof PostAggregation) ?
+                Collections.emptySet() : ImmutableSet.of((PostAggregation) sourceMetric);
```

Comment:
I would flip these conditions, and make the check a positive check rather than a negative one.